### ISSUE TITLE
[Discussion] Downloading the install script over an HTTP connection is a bad idea

### ISSUE THREAD

**Issue by N0ps32** [oh-my-fish/oh-my-fish#482]
The user is executing untrusted code that has been transferred over an insecure connection. Well technically only the redirect is insecure but since it's running over HTTP it's extremely easy to execute a MITM attack here and potentially execute harmful code by changing the redirect destination.

I think a good example of how this should be done is composer: https://getcomposer.org/download/

The script is:

- Always transferred over an HTTPS connection
- Verified with a hash once it's downloaded

Considering the "get.oh-my.fish" domain only returns a redirect anyway wouldn't it be better to just directly include the secure githubusercontent URL? Or at least verify the script integrity with a hash before executing it. I know the command won't look as "sexy" and short as it does now but it would really help improve security.

(or just get a free SSL certificate and migrate to HTTPS)

**Comment by oranja** [comment]
We have a SSL certificate for _oh-my.fish_, but when I just set the domain up I opted to let the DNS provider handle the simple redirect. My personal unfamiliarity with `nginx` and infrastructure management in general sounded like an equally dangerous risk factor.
I agree that the insecure HTTP redirect should be gone.

As an added value, we then would be able to count downloads and also might get rid of the recent nginx errors that we are getting from the DNS and redirect provider.

The hash verification would also be a great addition, and the more streamlined (mandatory but not interrupting), the merrier.

My only caveat is the "extremely easy" part. I agree that it's extremely possible, but what makes you say that it's extremely easy to implement in practice? I'm asking so that everyone would get a better understanding of the threat model.
